### ISSUE TITLE
Weekly View Feedback

### ISSUE THREAD

**Issue by ocdtrekkie** [zenhack/sandcal#28]
As per previous notes, I'm aware this is at an earlier stage of development, but I wanted to share my specific/actionable thoughts, in case it provides helpful ideas on approaches for improvement:

1. The weekly view should display the range of dates it is displaying at the top.
2. When I have a calendar item at 2:00 PM, for example, 2:00 PM does not display for that day, and I have to infer where the meeting is by surrounding squares. Headings should not disappear because events are present.
3. Similarly, if I have a meeting in partway through an hour, I have no visible reference to the start time. 2:00 PM displays, and my 2:30 is in the empty box between 2:30 and 3:00. Events should begin with their exact start time.
4. The width of the week on my screen seems to change in seemingly random ways based on the contents of items on my schedule.
5. I have to scroll down to the part of the day I normally care about (9 AM to 9 PM or so), because the view starts at midnight. I would either maybe start this view/autoscroll it to/auto-hide anything above 6 AM, or for more dynamic behavior, display time starting at the earliest appointment on the schedule.
6. Possibly narrow rows where no events exist at a given time all week to help reduce scrolling, as weekly view is presently like three desktop screens tall, even when only one or two events are present.

**Comment by zenhack** [comment]
Thanks for the feedback!

Yeah, a lot of this is stuff I'm aware of/already planning on fixing, but I wanted to get the MVP out the door.

Re: (4), can you show me some examples?

**Comment by ocdtrekkie** [comment]
I will email you a link to a video recording.

**Comment by zenhack** [comment]
I have taken the liberty of editing your comment to be a checklist.

**Comment by zenhack** [comment]
Related to the width changing (which is fixed on master), I also noticed the height of the cells sometimes changes, when you have an event with a long title. Probably should pick a sensible size and just cut it off.

**Comment by ocdtrekkie** [comment]
> Related to the width changing (which is fixed on master), I also noticed the height of the cells sometimes changes, when you have an event with a long title. Probably should pick a sensible size and just cut it off.

It probably should cut off the event title at maybe two rows, as opposed to the cell height. It's plausible someone may have multiple items in a row, and it may be well warranted for the cell height to grow to accommodate that.

**Comment by ocdtrekkie** [comment]
We are now *very literal* about taking all available space in 0.8:

![image](https://user-images.githubusercontent.com/4399499/190621040-4dfe8a63-60c8-4143-afd6-584ef7d644d4.png)

**Comment by ocdtrekkie** [comment]
Additionally, "all day" events currently just obliterate the whole day, it looks like. Other than that, I like that events now expand to their end times.

![image](https://user-images.githubusercontent.com/4399499/190622077-ffd58d5b-34b3-4483-a860-7659fd562241.png)

**Comment by zenhack** [comment]
Quoting Jacob Weisz (2022-09-16 06:39:25)

>    We are now very literal about taking all available space in 0.8:

Can you explain what I'm looking at here/what the problem is? Is this on
mobile? It scales correctly for me...

**Comment by ocdtrekkie** [comment]
That was desktop Firefox, it appears to go slightly wider than the iframe for some reason.

**Comment by zenhack** [comment]
Interesting, Firefox doesn't do that for me... I suppose I could add a small margin.

**Comment by zenhack** [comment]
Pushed a small update that introduces a margin; want to give it a try?

**Comment by ocdtrekkie** [comment]
Looks good to me!

**Comment by zenhack** [comment]
Still on the TODO list is making it deal with multiple events on the
same day; my thinking was in that case to squeeze the events next to
each other side-by-side. Once that's in place and there are timestamps
on the events themselves, my feeling is this will be what you want --
the event is all day. Thoughts?

Quoting Jacob Weisz (2022-09-16 06:46:33)

**Comment by ocdtrekkie** [comment]
Yeah that sounds fine. The only other things on here then would be trying to get the view to more easily focus on the relevant/interesting part of the day, and the handling of All Day events, which shouldn't normally block out the day, I think?
